File: src/core/abstraction/data/ISyncable_Repository.ts
import { ISyncMetadata } from "../metadata/ISync_metadata";
import { IBaseRepository } from "./base_repository";

export interface ISyncableRepository<T> extends IBaseRepository<T> {
  updateMany(entities: T[], metadata: ISyncMetadata): Promise<T[]>;
  addMany(entities: T[], metadata: ISyncMetadata): Promise<T[]>;
  removeMany(entities: T[], metadata: ISyncMetadata): Promise<T[]>;
  fetchMany(metadata: ISyncMetadata):Promise<T[]>;
}

File: fast_sync_dotnet_service/fast-sync-core/abstraction/data/ISyncableRepository.cs
using fast_sync_core.implementation.data;

namespace fast_sync_core.abstraction.data
{
    public interface ISyncableRepository<T> : IBaseRepository<T>
         where T : SyncableObject
    {
        Task<List<T>> UpdateMany(List<object> jsonEntities, ISyncMetadata metadata);
        Task<List<T>> AddMany(List<object> jsonEntities, ISyncMetadata metadata);
        Task<List<T>> RemoveMany(List<object> jsonEntities, ISyncMetadata metadata);
        Task<List<T>> FetchMany(ISyncMetadata jsonEntities);
    }
}
